lib/doorkeeper_auth.rb

```diff
@@ -0,0 +1,28 @@
+module DoorkeeperAuth
+  class DisallowedAccessError < StandardError; end
+
+  def self.included(base)
+    base.extend ClassMethods
+
+    base.class_attribute :api_accessible
+    base.before_action :access_denied?
+  end
+
+  private
+
+  def access_denied?
```

Comment:
Based on its name I would expect this method to return a boolean, not raise an error.